src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -148,10 +149,9 @@
                         "jruby-puppet.max-active-instances.")}))
      (when (jruby-schemas/shutdown-poison-pill? pool-instance#)
        (jruby/return-instance ~jruby-service pool-instance# ~reason)
-       (sling/throw+
-        {:type    ::service-unavailable
-         :message (str "Attempted to borrow a JRuby instance from the pool "
-                       "during a shutdown. Please try again.")}))
+       (mw/throw-service-unavailable!
```

Comment:
I don't know how much ceremony is around z's but if it was a pretty lightweight process I'd be down to put up a maint PR that duplicated these helpers into a utils namespace and deprecated the ones in the core namespace this afternoon.